src/EditorFeatures/CSharp/SplitStringLiteral/SplitStringLiteralCommandHandler.cs

```diff
@@ -98,41 +114,41 @@ public bool ExecuteCommandWorker(ReturnKeyCommandArgs args)
             return true;
         }
 
-        private bool SplitString(ITextView textView, ITextBuffer subjectBuffer, SnapshotPoint caret)
+        private bool SplitString(Document document, ITextView textView, ITextBuffer subjectBuffer, int position, CancellationToken cancellationToken)
         {
-            var document = subjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+            using var transaction = CaretPreservingEditTransaction.TryCreate(
+                CSharpEditorResources.Split_string, textView, _undoHistoryRegistry, _editorOperationsFactoryService);
+
+            var useTabs = !textView.Options.IsConvertTabsToSpacesEnabled();
+            var tabSize = textView.Options.GetTabSize();
+
+            // TODO: read option from textView.Options (https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1412138)
+            var indentStyle = document.Project.Solution.Options.GetOption(FormattingOptions.SmartIndent, LanguageNames.CSharp);
 
-            if (document != null)
+            var splitter = StringSplitter.TryCreate(document, position, useTabs, tabSize, indentStyle, cancellationToken);
+            if (splitter?.TrySplit(out var newDocument, out var newPosition) != true)
             {
-                var options = document.GetOptionsAsync(CancellationToken.None).WaitAndGetResult(CancellationToken.None);
-                var enabled = options.GetOption(SplitStringLiteralOptions.Enabled);
+                return false;
+            }
 
-                if (enabled)
-                {
-                    using var transaction = CaretPreservingEditTransaction.TryCreate(
-                        CSharpEditorResources.Split_string, textView, _undoHistoryRegistry, _editorOperationsFactoryService);
-
-                    var cursorPosition = SplitStringLiteral(document, options, caret, CancellationToken.None);
-                    if (cursorPosition != null)
-                    {
-                        var snapshotPoint = new SnapshotPoint(
-                            subjectBuffer.CurrentSnapshot, cursorPosition.Value);
-                        var newCaretPoint = textView.BufferGraph.MapUpToBuffer(
-                            snapshotPoint, PointTrackingMode.Negative, PositionAffinity.Predecessor,
-                            textView.TextBuffer);
-
-                        if (newCaretPoint != null)
-                        {
-                            textView.Caret.MoveTo(newCaretPoint.Value);
-                        }
-
-                        transaction.Complete();
-                        return true;
-                    }
-                }
+            // apply the change:
+            var workspace = newDocument.Project.Solution.Workspace;
+            workspace.TryApplyChanges(newDocument.Project.Solution);
```

Comment:
interesting that we're applying this as a workspace change.  is that what we did before?

Reply:
yup.  that's what we did before.